### PR TITLE
Update bundler-audit vulnerability database as app user

### PR DESCRIPTION
**Pull request by dblandin**
We're currently installing gems and updating the `bundler-audit` vulnerability database as the `root` user.

`bundle-audit update` downloads the database relative to the current user's home directory. As the `root` user, the database is stored at `/root/.local/share/ruby-advisory-db`.

Following best security practices, we drop down to a less privileged user within our `Dockerfile`.

Like the `bundle-audit update` command, `bundle-audit check` will refer to a vulnerability database relative to the current user's home directory. Running as the `app` user, `bundler-audit` looks for a database at `/home/app/.local/share/ruby-advisory-db`.

So, to summarize, we've been downloading the vulnerability database to `/root/.local/share/ruby-advisory-db` and checking for it at `/home/app/.local/share/ruby-advisory-db` during engine runs. This obviously doesn't work and consistently produced out-of-date analysis results.

This PR updates this project's `Dockerfile` to update the `bundler-audit` vulnerability database after we switch down to the `app` user, so it goes to the right directory and can be picked up by the running engine.

@codeclimate/review :mag_right:

### CODE REVIEW

**Comment by jpignata** [comment]
LGTM

**Comment by dblandin** [comment]
@codeclimate/review, @jp This is ready for another :mag_right:

**Comment by jpignata** [comment]
:+1:
